src/python/pants/base/workunit.py

```diff
@@ -246,6 +246,18 @@ def root(self):
       ret = ret.parent
     return ret
 
+  def is_background(self, background_root_workunit):
```

Comment:
I recommend using a keyword only arg here, so that callers must use a kwarg.

```suggestion
  def is_background(self, *, background_root_workunit):
```

Reply:
This is a required parameter of a particular type, rather than a boolean or optional parameter. So I'm not sure I agree in this case.

But @cattibrie : it would be good to add a docstring for this method.

Reply:
@Eric-Arellano Is it likely to ever accept more than one argument? I think that if not (it seems not), then a positional param seems ok.

Reply:
Eh after typing up the below example, I agree that yes it's fine to not use kwarg only here:

```python
lint_workunit.is_background(compile_workunit)
lint_workunit.is_background(background_root_workunit=compile_workunit)
```

--

However, in general, want to respond to this:

> This is a required parameter of a particular type, rather than a boolean or optional parameter.

True, but there is often (not always) value in enforcing kwarg only for required parameters too. For example, from `strutil.py`:


```python
def strip_prefix(string: str, prefix: str) -> str: ...

strip_prefix("hello world", "hello")
strip_prefix("hello world", prefix="hello")
```

You can of course figure it out with the first version, but the second has no ambiguity and I would argue is worth the extra few characters.

Reply:
@Eric-Arellano : If the arguments had different types, explicitly naming them would be slightly less valuable though: that's what I was referring to here.